snack-shop-fe/src/App.tsx

```diff
@@ -1,35 +1,54 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { useEffect, useState } from 'react';
 
-function App() {
-  const [count, setCount] = useState(0)
+const App = () => {
+  const [data, setData] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const fetchDataForSnacks = async () => {
+      try {
+        const response = await fetch(
+          `http://localhost:3000/snacks`
+        );
+        if (!response.ok) {
```

Comment:
We should consider introducing custom error handling for network connection errors as well so we have full control what's provided to the user and that it doesn't leak valuable information for malicious actors, like tech stack or domain insights. Right now, what's provided to the user in case of network connection errors is decided by the browser. E.g. in case bff is down, FireFox shows "NetworkError when attempting to fetch resource." while Chrome says "Failed to fetch".

We can either do it now already, or keep it in mind for later when we have more insights on how the implementation evolved.

Reply:
A great idea for the backlog, thank you.